src/NServiceBus.Transport.Msmq/DelayedDelivery/DelayedMessage.cs

```diff
@@ -0,0 +1,41 @@
+
+namespace NServiceBus
+{
+    using System;
+
+    /// <summary>
+    /// Represents a delayed message.
+    /// </summary>
+    public class DelayedMessage
+    {
+        /// <summary>
+        /// Date and time the message is due.
+        /// </summary>
+        public DateTime Time { get; set; }
+
+        /// <summary>
+        /// Native message ID
+        /// </summary>
+        public string Id { get; set; }
```

Comment:
Now thinking about it, should this be `TransportMessageId` or `MessageId`?
```suggestion
        public string Id { get; set; }
```

Where would be want to apply idempotency? I think transport message ID is safest. Although that could result in having multiple "clones" with the same message ID but different transport ID's.

Reply:
MessageId seems to make more sense to me, especially to avoid duplicates?